Sources/UIKit/PortalCollectionView.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  PortalCollectionView.swift
+//  PortalView
+//
+//  Created by Argentino Ducret on 4/4/17.
+//  Copyright © 2017 Guido Marucci Blas. All rights reserved.
+//
+
+import UIKit
+
+public final class PortalCollectionView<MessageType>: UICollectionView, UICollectionViewDataSource, UICollectionViewDelegate {
+    
+    public let mailbox = Mailbox<MessageType>()
+    public var isDebugModeEnabled: Bool = false
+    public var isSnapToCellEnabled: Bool = false
+    
+    fileprivate let layoutEngine: LayoutEngine
+    fileprivate let items: [CollectionItemProperties<MessageType>]
+    fileprivate var selected: Int = 0
+    fileprivate var lastOffset: CGFloat = 0
+    
+    public init(items: [CollectionItemProperties<MessageType>], layoutEngine: LayoutEngine, layoutValues: CollectionViewLayoutValues) {
+        self.items = items
+        self.layoutEngine = layoutEngine
+        let layout = createFlowLayout(layoutValues: layoutValues)
+        super.init(frame: .zero, collectionViewLayout: layout)
+   
+        self.dataSource = self
+        self.delegate = self
+        
+        let identifiers = Array(Set(items.map { $0.identifier }))
```

Comment:
I don't think is necessary to convert the set to an array just to iterate over the elements.